test/integration/migrate/index.js

```diff
@@ -268,12 +221,52 @@ module.exports = function(knex) {
 
   });
 
+  describe('knex.migrate.latest in parallel', function() {
+
+    afterEach(function() {
+      return knex.migrate.rollback({directory: 'test/integration/migrate/test'});
+    });
+
+    if (knex.client.dialect === 'postgres') {
+      it("is able to run two migrations in parallel in postgres", function () {
```

Comment:
I just tested this with mysql and it should work the same way that postgres does. So there is no reason why this test wouldn't work for every other DB (except sqlite).

![mysql-selectforupdate](https://cloud.githubusercontent.com/assets/696345/24333090/de047930-125a-11e7-9767-6bda3ff59698.gif)